cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java

```diff
@@ -164,9 +180,9 @@ public void validateTags(Id.NamespacedId entityId, String ... tags) throws Inval
 
   // Validate the key should not be be reserved {@link BusinessMetadataDataset.TAGS_KEY}
   private void validateTagReservedKey(Id.NamespacedId entityId, String key) throws InvalidMetadataException {
-    if (BusinessMetadataDataset.TAGS_KEY.equals(key.toLowerCase())) {
+    if (MetadataDataset.TAGS_KEY.equals(key.toLowerCase())) {
```

Comment:
Good point regarding tags. What I'm unsure of at this point is how we will depict this in the UI. If the UI will do one call and expect a set of tags across SYSTEM and USER scopes, then I think we may need some special handling. Otherwise, we should just group the output of getAllTags by SCOPE, and let clients (UI, CLI) handle them separately.